### ISSUE TITLE
Xamarin Studio crashes when loading a project

### ISSUE THREAD

**Issue by DamianSuess** [picoe/Eto#869]
After enabling Eto.Forms add-in on Xamarin Studio I'm getting the following error when attempting to open any C# solution. The current workaround is to disable the Eto.Forms add-in.

Has anyone else encountered this error?  Any thoughts or tips on how I could assist in debugging this issue would be greatly appreciated.  If you need a complete log file, I could get that.

## Environment:
* **IDE:** Xamarin Studio v6.3 (build 863) 
* **Addin:** Eto.Forms Support Addin v2.3.0.6
* **OS:** Windows 10

## XS's log, "Ide.2017-08-23__13-28-05.log"
```
FATAL ERROR [2017-08-23 12:39:30Z]: An unhandled exception has occured. Terminating Xamarin Studio? True
System.InvalidCastException: [A]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions cannot be cast to [B]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions. Type A originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'LoadFrom' at location 'C:\Users\USER_NAME\AppData\Local\XamarinStudio-6.0\LocalInstall\Addins\Eto.Addin.XamarinStudio.2.3.0.6\Microsoft.CodeAnalysis.CSharp.dll'. Type B originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'Default' at location 'C:\Program Files (x86)\Xamarin Studio\bin\Microsoft.CodeAnalysis.CSharp.dll'.
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTreeFactoryServiceFactory.CSharpSyntaxTreeFactoryService.ParseSyntaxTree(String fileName, ParseOptions options, SourceText text, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DocumentState.<FullyParseTreeAsync>d__11.MoveNext()
FATAL ERROR [2017-08-23 12:39:30Z]: An unhandled exception has occured. Terminating Xamarin Studio? True
System.InvalidCastException: [A]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions cannot be cast to [B]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions. Type A originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'LoadFrom' at location 'C:\Users\USER_NAME\AppData\Local\XamarinStudio-6.0\LocalInstall\Addins\Eto.Addin.XamarinStudio.2.3.0.6\Microsoft.CodeAnalysis.CSharp.dll'. Type B originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'Default' at location 'C:\Program Files (x86)\Xamarin Studio\bin\Microsoft.CodeAnalysis.CSharp.dll'.
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTreeFactoryServiceFactory.CSharpSyntaxTreeFactoryService.ParseSyntaxTree(String fileName, ParseOptions options, SourceText text, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DocumentState.<FullyParseTreeAsync>d__11.MoveNext()
FATAL ERROR [2017-08-23 12:39:30Z]: An unhandled exception has occured. Terminating Xamarin Studio? True
System.InvalidCastException: [A]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions cannot be cast to [B]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions. Type A originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'LoadFrom' at location 'C:\Users\USER_NAME\AppData\Local\XamarinStudio-6.0\LocalInstall\Addins\Eto.Addin.XamarinStudio.2.3.0.6\Microsoft.CodeAnalysis.CSharp.dll'. Type B originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'Default' at location 'C:\Program Files (x86)\Xamarin Studio\bin\Microsoft.CodeAnalysis.CSharp.dll'.
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTreeFactoryServiceFactory.CSharpSyntaxTreeFactoryService.ParseSyntaxTree(String fileName, ParseOptions options, SourceText text, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DocumentState.<FullyParseTreeAsync>d__11.MoveNext()
FATAL ERROR [2017-08-23 12:39:30Z]: An unhandled exception has occured. Terminating Xamarin Studio? True
System.InvalidCastException: [A]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions cannot be cast to [B]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions. Type A originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'LoadFrom' at location 'C:\Users\USER_NAME\AppData\Local\XamarinStudio-6.0\LocalInstall\Addins\Eto.Addin.XamarinStudio.2.3.0.6\Microsoft.CodeAnalysis.CSharp.dll'. Type B originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'Default' at location 'C:\Program Files (x86)\Xamarin Studio\bin\Microsoft.CodeAnalysis.CSharp.dll'.
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTreeFactoryServiceFactory.CSharpSyntaxTreeFactoryService.ParseSyntaxTree(String fileName, ParseOptions options, SourceText text, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DocumentState.<FullyParseTreeAsync>d__11.MoveNext()
ERROR [2017-08-23 12:41:16Z]: Connection ping failed
MonoDevelop.Core.Execution.RemoteProcessException: Not connected
   at MonoDevelop.Core.Execution.RemoteProcessConnection.PostMessage(BinaryMessage message, TaskCompletionSource`1 cs, Boolean checkInitialized) in C:\data\lanes\4469\a2163670\source\monodevelop\main\src\core\MonoDevelop.Core\MonoDevelop.Core.Execution\RemoteProcessConnection.cs:line 333
   at MonoDevelop.Core.Execution.RemoteProcessConnection.SendMessage(BinaryMessage message) in C:\data\lanes\4469\a2163670\source\monodevelop\main\src\core\MonoDevelop.Core\MonoDevelop.Core.Execution\RemoteProcessConnection.cs:line 310
   at MonoDevelop.Core.Execution.RemoteProcessConnection.PingConnection(Object state) in C:\data\lanes\4469\a2163670\source\monodevelop\main\src\core\MonoDevelop.Core\MonoDevelop.Core.Execution\RemoteProcessConnection.cs:line 656
FATAL ERROR [2017-08-23 12:41:16Z]: An unhandled exception has occured. Terminating Xamarin Studio? True
System.InvalidCastException: [A]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions cannot be cast to [B]Microsoft.CodeAnalysis.CSharp.CSharpParseOptions. Type A originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'LoadFrom' at location 'C:\Users\USER_NAME\AppData\Local\XamarinStudio-6.0\LocalInstall\Addins\Eto.Addin.XamarinStudio.2.3.0.6\Microsoft.CodeAnalysis.CSharp.dll'. Type B originates from 'Microsoft.CodeAnalysis.CSharp, Version=1.3.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35' in the context 'Default' at location 'C:\Program Files (x86)\Xamarin Studio\bin\Microsoft.CodeAnalysis.CSharp.dll'.
   at Microsoft.CodeAnalysis.CSharp.CSharpSyntaxTreeFactoryServiceFactory.CSharpSyntaxTreeFactoryService.ParseSyntaxTree(String fileName, ParseOptions options, SourceText text, CancellationToken cancellationToken)
   at Microsoft.CodeAnalysis.DocumentState.<FullyParseTreeAsync>d__11.MoveNext()
```

**Comment by Linchy** [comment]
I'm getting issues compiling, since upgrading Xamarin to 6.3:

![image](https://user-images.githubusercontent.com/1840551/29685380-38784d14-890d-11e7-9ab8-d36f86f9f9c8.png)

Seems to not support C# 6 features anymore.

**Comment by Linchy** [comment]
Oh fixed my issue - opened solution up in VS and set the platform target from x86 to Any CPU. @DamianSuess you may want to try opening in VS as well, and see if you can get it to compile successfully there first.

**Comment by DamianSuess** [comment]
@Linchy, thank you. I'll give it a shot

**Comment by cwensley** [comment]
Thanks for the report!  This should be fixed when the addin is updated for VS/MD 7.x.  I won't be supporting MD 6.x in the future unfortunately.  I believe the MD 6.x addin only works on macOS currently.  However, on  windows there is an addin for VS 2015 that works just as well.  I am also updating this for VS 2017 at the moment.

**Comment by cwensley** [comment]
Should now be fixed with #974.  A new version of the addin will be published along with the 2.4 release, or you can build it manually.
